### PR TITLE
WIP: add codspeed benchmarks [skip cirrus]

### PR DESCRIPTION
**Pull request by ev-br**
cross-ref https://github.com/ev-br/openblas-bench/issues/6

### CODE REVIEW

File: .github/workflows/codspeed-bench.yml [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+          cd build/openblas_wrap
+          python -c'import _flapack; print(dir(_flapack))'
+
+      - name: Check the wrapper is importable
+        run: |
+          cd benchmark/pybench
+          export PYTHONPATH=$PWD/build-install/lib/python${{matrix.pyver}}/site-packages/
+          echo "PYTHONPATH = "$PYTHONPATH
+          ls -l $PYTHONPATH
+          ls -l $PYTHONPATH/openblas_wrap
```
Comment (ev-br) [inline .github/workflows/codspeed-bench.yml]:
```suggestion
          ls -l $PYTHONPATH/openblas_wrap
          cd $PYTHONPATH/openblas_wrap
          python -c'import _flapack; print(dir(_flapack))'
```

**Comment by ev-br** [comment]
Phew. Finally managed to make it run on codspeed. Will clean up the codspeed action and revert back the other CI.

However, https://github.com/OpenMathLib/OpenBLAS/actions/runs/9138697348/job/25130179181?pr=4678 reports that

```
Upload the results
  CodSpeed Run Hash: "47ae471c56db8c1163d892307489c4ea202671cd8d4197ca08586f9ab23bddfd"
  Preparing upload...
  Error: Error Failed to retrieve upload data: 401 Unauthorized {"error":"The repository is not enabled"}
  Error: Process completed with exit code 1.
```

WDYT @martin-frbg ?

**Comment by martin-frbg** [comment]
Idk, this repository has never worn a codpiece before. Their site has now asked me to enable the repo and add a codspeed_token (which I don`t think it did before) so maybe it will work now

**Comment by ev-br** [comment]
Yup, it did work this time: https://codspeed.io/OpenMathLib/OpenBLAS/branches/ev-br:codspeed !
The PR is ready from my side.

**Pull request merged**